tests/test_interface.py

```diff
@@ -16,4 +17,92 @@ def test_matrix_multiplication(a, b):
 
     expected = np.matmul(a, b)
     
-    assert_equal(result, expected)
\ No newline at end of file
+    assert_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "a",
+    [
+        (np.array([2, 3, 4])),
+    ],
+)
+def test_vector_prod(a):
+    print("***** prod *****")
+    
+    la = finch.lazy(a)
+    result = finch.prod(la, -1)
```

Comment:
you must call `compute` on the resulting lazy tensor to get an actual tensor result